packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -793,6 +823,42 @@ export class Vector2 {
     }
 }
 
+/**
+ * Class used to store (r, theta) vector representation
+ */
+export class Polar {
+	public length: number;
+	public theta: number;
+
+	/**
+	 * Creates a new Polar object
+	 * @param length the length of the vector
+	 * @param theta the angle of the vector
+	 */
+	constructor(length: number, theta: number){
+		this.length = length;
+		this.theta = theta;
+	}
+
+	/**
+	 * Gets the rectangular coordinates of the current Polar
+	 * @returns the rectangular coordinates
+	 */
+	public toVector2(): Vector2{
+		return Vector2.fromPolar(this);
+	}
+
+	/**
+	 * Converts a given Vector2 to its polar coordinates
+	 * @param v the Vector2 to convert
+	 * @returns a Polar
+	 */
+	public static fromVector2(v: Vector2): Polar{

```

Comment:
Static so Pascal Case need.

Does not give correct values for theta in all cases - see https://playground.babylonjs.com/#ZUFVG7

For correct values of theta see - https://playground.babylonjs.com/#ZUFVG7#1
no need for GetAngleBetweenVectors